### PR TITLE
File sharing 

### LINKED ISSUE #5: [Feature 4.2] File Sharing Conversion and Sending
Converting root address to URIs for file sharing, then pass the URIs out to make a new Intent, and send using Activity. Notify directory to switch view when finished.

### LINKED ISSUE #13: [Feature 4.1] File Sharing Receiving
Obtaining and passing a list of root addresses from the directory.

### PR DESCRIPTION
**Pull request by JerryLi-617**
Closes #5 
Closes #13

Added classes for receiving IO.Files to convert to content type Uri, and using android sharesheet to share said files. 

All Java classes added are within the file_sharing folder. 
Include a new XML file named file_paths.xml in the res.xml folder for file path access for FileProvider.
Include changes to AndroidManifest.xml to implement FileProvider.

Not yet linked to Directory, this feature will not show when running the app. Changes should not have effects on running the app on an emulator.

### CODE REVIEW

**Comment by CSEndeavour** [comment]
Add comments to other use cases.

**Comment by CSEndeavour** [comment]
For the usecases, remove the getContext() methods since Android Studio has it's own.

**Review by CSEndeavour — APPROVED** [review]

File: app/src/main/java/com/example/notes2text/file_sharing/use_case/MyFileProvider.java
Comment (julianmarrero) [inline app/src/main/java/com/example/notes2text/file_sharing/use_case/MyFileProvider.java:1]: MyFileProvider does not seem to be called anywhere. What is the purpose of this class?

**Review by julianmarrero — APPROVED** [review]

File: app/src/main/java/com/example/notes2text/file_sharing/use_case/MyFileProvider.java
Comment (JerryLi-617) [reply app/src/main/java/com/example/notes2text/file_sharing/use_case/MyFileProvider.java:1]: Required implementation by Android to configure FileProvider. Will be adding documentation to make it clearer.

File: app/src/main/java/com/example/notes2text/file_sharing/use_case/ThirdPartyOutputModel.java
Comment (ChowdhuryShaswata) [inline app/src/main/java/com/example/notes2text/file_sharing/use_case/ThirdPartyOutputModel.java:1]: Add documentation to the output model - particularly, to the class and to all the public methods. Documentation is needed for any publicly accessible objects.

File: app/src/main/java/com/example/notes2text/file_sharing/use_case/MyFileProvider.java
Comment (ChowdhuryShaswata) [inline app/src/main/java/com/example/notes2text/file_sharing/use_case/MyFileProvider.java:1]: In addition to adding documentation, consider renaming the class to be more evocative than MyFileProvider - class name should quickly reflect usage.

**Review by ChowdhuryShaswata — COMMENTED** [review]
I don't see the FileMenuInteractor class among the list of changed files. Have you had the chance to add in the connection point to the directory?

File: app/src/main/java/com/example/notes2text/file_sharing/use_case/ThirdParShareUseCase.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package com.example.notes2text.file_sharing.use_case;
+
+import java.util.ArrayList;
+import java.io.File;
+
+import android.net.Uri;
+import androidx.core.content.FileProvider;
+
+
+public class ThirdParShareUseCase implements ThirdParShareInputBoundary {
```
Comment (ChowdhuryShaswata) [inline app/src/main/java/com/example/notes2text/file_sharing/use_case/ThirdParShareUseCase.java]: You should document the class as well.

File: app/src/main/java/com/example/notes2text/file_sharing/use_case/ShareObserver.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package com.example.notes2text.file_sharing.use_case;
+
+import com.example.notes2text.file_sharing.adapters.ThirdPartySender;
+
+import android.content.Context;
+
+import java.io.File;
+import java.util.ArrayList;
+
+// Used to better accommodate to Android workflow. The method has same role as main.java.
```
Comment (ChowdhuryShaswata) [inline app/src/main/java/com/example/notes2text/file_sharing/use_case/ShareObserver.java:10]: Mention for course purposes why this was preferred over input boundary.

File: app/src/main/java/com/example/notes2text/file_sharing/adapters/ThirdPartySender.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import android.content.Intent;
+
+
+public class ThirdPartySender implements ThirdPartyOutputBoundary {
+
+
+    /**
+     * Using the list of content Uri in the given outputModel, create a new Intent
+     * to access the share sheet function. Only sends text file of any type. OutputModel
+     * is returned so that context may be used to check the results of the action.
```
Comment (ChowdhuryShaswata) [inline app/src/main/java/com/example/notes2text/file_sharing/adapters/ThirdPartySender.java:18]: This method does not return output model - if this is intentional, please change your docstring to reflect that the output model is not returned.

File: app/src/main/java/com/example/notes2text/file_sharing/use_case/FileSharingModel.java
Comment (ChowdhuryShaswata) [inline app/src/main/java/com/example/notes2text/file_sharing/use_case/FileSharingModel.java:1]: Add in documentation.

**Review by ChowdhuryShaswata — APPROVED** [review]

**Pull request merged**